network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/exclusion/ui/NetpAppExclusionListActivity.kt

```diff
@@ -0,0 +1,241 @@
+                    enabled: Boolean,
+                    position: Int,
+                ) {
+                    viewModel.onAppProtectionChanged(app, enabled)
+                }
+
+                override fun onFilterClick(anchorView: View) {
+                    showFilterPopupMenu(anchorView)
+                }
+            },
+        )
+
+        val recyclerView = binding.netpAppExclusionListRecycler
+
+        if (isNetpEnabled) {
+            recyclerView.alpha = 1.0f
+        } else {
+            recyclerView.alpha = 0.45f
+        }
+
+        recyclerView.adapter = adapter
+    }
+
+    private fun showFilterPopupMenu(anchor: View) {
+        val popupMenu = PopupMenu(layoutInflater, R.layout.popup_exclusion_list_filter)
+        val view = popupMenu.contentView
+        val allItemView = view.findViewById<View>(R.id.exclusion_list_filter_all)
+        val protectedItemView = view.findViewById<View>(R.id.exclusion_list_filter_protected)
+        val unprotectedItemView = view.findViewById<View>(R.id.exclusion_list_filter_unprotected)
+
+        popupMenu.apply {
+            onMenuItemClicked(allItemView) { viewModel.applyAppsFilter(AppsFilter.ALL) }
+            onMenuItemClicked(protectedItemView) { viewModel.applyAppsFilter(AppsFilter.PROTECTED_ONLY) }
+            onMenuItemClicked(unprotectedItemView) { viewModel.applyAppsFilter(AppsFilter.UNPROTECTED_ONLY) }
+        }
+        popupMenu.showAnchoredToView(binding.root, anchor)
+    }
+
+    private fun renderViewState(viewState: ViewState) {
+        shimmerLayout.stopShimmer()
+        adapter.update(viewState, isNetpEnabled)
+        shimmerLayout.gone()
+    }
+
+    private fun processCommand(command: Command) {
+        when (command) {
+            is Command.RestartVpn -> restartVpn()
+            is Command.ShowDisableProtectionDialog -> showDisableProtectionDialog(
+                command.forApp,
+            )
+            is Command.ShowIssueReportingPage -> globalActivityStarter.start(this, command.params)
+            else -> { /* noop */
+            }
+        }
+    }
+
+    private fun restartVpn() {
+        // we use the app coroutine scope to ensure this call outlives the Activity
+        appCoroutineScope.launch(dispatcherProvider.io()) {
+            vpnFeaturesRegistry.refreshFeature(NetPVpnFeature.NETP_VPN)
+        }
+    }
+
+    private fun showDisableProtectionDialog(app: NetpExclusionListApp) {
+        val dialog = ManuallyDisableAppProtectionDialog.instance(app)
+        dialog.show(
+            supportFragmentManager,
+            ManuallyDisableAppProtectionDialog.TAG_MANUALLY_EXCLUDE_APPS_DISABLE,
+        )
+    }
+
+    companion object {
+        enum class AppsFilter {
+            ALL,
+            PROTECTED_ONLY,
+            UNPROTECTED_ONLY,
+        }
+
+        internal fun intent(context: Context): Intent {
+            return Intent(context, NetpAppExclusionListActivity::class.java)
+        }
```

Comment:
Maybe we can use the `GlobalActivityStarter` instead of the old static `intent` pattern

Reply:
I've addressed this in the treatment PR. 👍